opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -86,94 +90,142 @@ message InstrumentationLibraryMetrics {
-    INT64 = 1;
+    // ScalarInt64 implies that the value is found in
+    // Metric.scalar_data_points[].value_int64.
+    SCALAR_INT64 = 1;
 
-    // MONOTONIC_INT64 values are monotonically increasing signed 64-bit
-    // integers.
-    //
-    // A Metric of this Type MUST store its values as Int64DataPoint.
-    MONOTONIC_INT64 = 2;
+    // ScalarDouble implies that the value is found in
+    // Metric.scalar_data_points[].value_double.
+    SCALAR_DOUBLE = 2;
 
-    // DOUBLE values are double-precision floating-point numbers.
-    //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    DOUBLE = 3;
+    // Histogram implies that the value is found in
+    // Metric.histogram_data_points[].histogram.
+    HISTOGRAM = 3;
 
-    // MONOTONIC_DOUBLE values are monotonically increasing double-precision
-    // floating-point numbers.
-    //
-    // A Metric of this Type MUST store its values as DoubleDataPoint.
-    MONOTONIC_DOUBLE = 4;
-
-    // Histogram measurement.
-    // Corresponding values are stored in HistogramDataPoint.
-    HISTOGRAM = 5;
-
-    // Summary value. Some frameworks implemented Histograms as a summary of observations
-    // (usually things like request durations and response sizes). While it
-    // also provides a total count of observations and a sum of all observed
-    // values, it calculates configurable percentiles over a sliding time
-    // window.
-    // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 6;
+    // Summary implies that the value is found in
+    // Metric.summary_data_points[].summary.
+    SUMMARY = 4;
+
+    // TODO(#159): add RAW_INT64 and RAW_DOUBLE value types.
   }
 
-  // type is the type of values this metric has.
-  Type type = 4;
+  // ValueType is the type of values this metric has.
+  ValueType value_type = 4;
+
+  // KindElement contains a set of bit masks used to construct Kind
+  // enum values.  There are 7 bits used presently, broken into
+  // groups:
+  //
+  // "Temporality" is the temporal quality of a metric, indicating how
+  // values relate to the time interval over which they are reported.
+  // One of the 3 Temporality values MUST be set: CUMULATIVE, DELTA,
+  // or INSTANTANEOUS.
+  //
+  // "Structure" is the structural quality of a metric, indicating
+  // whether metric data describes a sum of measurements (ADDING) or a
+  // collection of individual measurements (GROUPING).  Value types
+  // may be interpreted differently depending on Structure.  For
+  // example, a Histogram DataPoint may be computed for a Counter
+  // (ADDING) instrument or a ValueRecorder (GROUPING) instrument, and
+  // the Sum and Count of the resulting aggregation may be interpreted
+  // differently depending on structure.
+  //
+  // MONOTONIC is a boolean option that may be applied to
+  // ADDING-structure metrics.
+  //
+  // ASYNCHRONOUS is a boolean option that may be applied to all
```

Comment:
Sorry that I am keep repeating this, but I fail to understand the benefit of this bit.

Reply:
"system knows not to display a rate of the collection events" -- Do you mean taking the rate of a metric collected on an interval like Otel observers "shouldn't" be done? I'm not sure that makes sense so maybe I'm misreading this.

Reply:
The "rate of a metric" is OK.

It's the "rate of events for a metric" (i.e., rate of the Count field for Histogram and Summary points), which is something that _is_ meaningful for an ordinary GROUPING metrics. This is where an ASYNCHRONOUS or not distinction can be used.

Reply:
Aaaah, got it.

Reply:
Apologies if these are dumb questions, but I need more clarification:

1. Is it possible to have a Histogram or Summary aggregation for an asynchronous instrument where you might want to calculate the rate of events? I'm trying to think of a case where this would be useful but can't think of any off hand.

2. In what case is it not true that we can compute meaningful ratios for a simple aggregation type? It seems like the same holds true for a simple example like the count of requests (e.g. `Method={GET} / Method={*}`)

Overall, the distinction between sync/async still feels a little tenuous to me as well. Maybe the term itself makes me a feel a little uncomfortable, as it doesn't really seem like consumers should care about _how_ a metric was generated.

Reply:
For question (1), it's possible to compute a histogram or summary aggregation of an asynchronous instrument. The raw events in this case are observations, and the API specifies that multiple observations for a single label set by the same callback are considered replacement values, so the count of raw events per label set is either 0 or 1 at the moment when an asynchronous instrument is evaluated. The number of raw events equals the cardinality of the set for a given evaluation of the callback. You could aggregate away the labels of these data points, combining them so that Count becomes greater than 1, still equal to the number of distinct label sets that were recorded. 

Now you could read all the data points for this instrument across a range of time, and the Count aggregation now computes something, but what is the meaning of this aggregation? This Count is a function of the collection period, because collecting more frequently yields more data points.  That is the property that ASYNCHRONOUS describes. The ASYNCHRONOUS bit does not change any interpretation or alter the mathematics of processing metric data, but it does indicate when a particular derived computation may be meaningless.

Reply:
For question (2), it's possible with a synchronous instrument to compute meaningful ratios (e.g. Method={GET} / Method={*}), but it requires a time interval to evaluate this query. The result depends on the time interval again, a small window will give you one result, and a large window gives another. The time variable has to be smoothed out of this equation, whereas with an ASYNCHRONOUS instrument the semantics of the instrument give us a better answer to this sort of query. I've sometimes referred to these "point-in-time" ratios.

Reply:
Also you're leading me to believe that ASYNCHRONOUS is not the best term here. I am thinking of alternatives, and am open to suggestion. The term originates from the synchronous usage: you call a synchronous instrument at a meaningful time, whereas the SDK calls asynchronous instruments at a meaningless time.

Reply:
Another term, possibly: 
NONCHRONOMETRIC

Reply:
Hmm okay, the point in time calculation def makes sense to me 👍

Re naming, wouldn't it just be CHRONOMETRIC? That name does sound more appealing than ASYNCHRONOUS imo.

Re #1, I'd still be interested to hear an actual example of a CHRONOMETRIC (asynchronously recorded) histogram.

Reply:
I'm not sure because CHRONOMETER refers to a stopwatch which is used for finely measuring time, and it feels like the synchronous instruments are more chronometrical than the asynchronous instruments. 

Another term I would support is NONCHRONOLOGICAL, since there is no meaningful order between events observed in the same callback. This seems to suggest more than the meaning given to ASYNCHRONOUS (NONCHRONOLOGICAL also suggests that the events of different callbacks are not order-able?).

Reply:
Hmm okay, do asynchronous callbacks always occur at a regular time interval or is that not necessarily the case? (that's what led me to believe they are "chronometric" although perhaps that doesn't mean what I thought it does)

Reply:
There is no specification about collection intervals being regular, and there is definitely interest in making it configurable. See https://github.com/open-telemetry/oteps/pull/121.

To summarize, I believe we've agreed there is something different about these metrics created by Observer instruments, but it's come to light that "Asynchronous" has been badly applied. The term CHRONOLOGICAL is appealing to me, now, and is one I'm willing to back-port into the API specification. That is to suggest we replace the existing SYNCHRONOUS / ASYNCHRONOUS distinction with a SYNCHRONOUS / CHRONOLOGICAL distinction for instruments.

Reply:
I replaced the use of ASYNCHRONOUS with CHRONOLOGICAL.

Reply:
Another possible suggestion: SNAPSHOT vs CONTINUOUS